drivers/cst816s/cst816s.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2020 Koen Zandberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_cst816s
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for cst816s touch screen
+ *
+ * @author      koen Zandberg <koen@bergzand.net>
+ *
+ * @}
+ */
+
+#include "log.h"
+#include "periph/gpio.h"
+#include "periph/i2c.h"
+#include "xtimer.h"
+
+#include "cst816s.h"
+#include "cst816s_internal.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+static void _gpio_irq(void *arg)
+{
+    cst816s_t *dev = arg;
+
+    dev->cb(dev, dev->cb_arg);
```

Comment:
I would only call this if `dev->cb` is not NULL (seem my previous comment in the init function).

```suggestion
    cst816s_t *dev = arg;
    assert(dev);

    if (dev->cb) {
        dev->cb(dev, dev->cb_arg);
    }
    else {
        (void)dev;
    }
```

Reply:
Adapted this without the `else` case